src/UCTNode.cpp

```diff
@@ -75,7 +75,6 @@ bool UCTNode::create_children(std::atomic<int>& nodecount,
     }
     // We'll be the one queueing this node for expansion, stop others
     m_is_expanding = true;
-    lock.unlock();
 
     const auto raw_netlist = Network::get_scored_moves(
```

Comment:
This takes a long time! I doubt we want to be locking over it.

Reply:
I think once the tree grows large it's not so likely that two threads hit the same unexpanded node simultaneously? We won't be able to go below that node anyway if the NN result hasn't been returned yet? You won't even have the policy.

Reply:
Agreed. Perhaps a way to solve it without extra locking time is to explicitly gate the search from going deeper if `m_net_eval` of the node is some initial value indicating it hasn't been populated yet?

Reply:
I am not sure what is the disadvantage of a lock that lasts long; I am not familiar with how it works. It should definitely halt a thread going into the node (that thread has to wait anyway), but does it affect other threads?

Reply:
I may have jumped to a conclusion too soon here. I was thinking this would block other threads relatively often but having a closer look I only can see that if another thread was exploring the expanding node's parent that it would have to wait e.g. in `uct_select_child()`

Reply:
OK I now think extending the lock isn't necessary after all. The mutex unlock when link_nodelist returns (after all the psa_ratio stuff is done) signals that create_children has finished and the new node is ready for uct_select_child (in play_simulation, has_children will be true); however, the new node may not have been updated with the NN eval. So it suffices to do the update before going into link_nodelist. I moved the unlock back, compiled and tested, and the crash no longer occurs.